File: warmup_1/Diff21.java
/**
 * Given an int n, return the absolute difference between n and 21, except return 
 * double the absolute difference if n is over 21.
 * 
 * <ul>
 *  <li>diff21(19) → 2
 *  <li>diff21(10) → 11
 *  <li>diff21(21) → 0
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p116624
 * @since 17.0.1
 * @version 0.0.2
 */
public class Diff21 {
    public static void main(String[] args) {
        System.out.println("diff21(19) -> " + diff21(19));
        System.out.println("diff21(10) -> " + diff21(10));
        System.out.println("diff21(21) -> " + diff21(21));
        System.out.println("diff21(20) -> " + diff21(20));
        System.out.println("diff21(100) -> " + diff21(100));
    }

    /**
     * Given an int n, return the absolute difference between n and 21, except return 
     * double the absolute difference if n is over 21.
     * 
     * @param n integer value that is being subtracted from 21
     * @return integer value that has been subtracted from 21 if the difference is positive,
     * integer value that has been subtracted from 21 and doubled if the difference is negative
     * @since 0.0.1
     */
    public static int diff21(int n) {
        if (n <= 21) {
            n = 21-n;
          } else {
            n = Math.abs(21-n) * 2;
          }
          return n;
    }
}

File: warmup_1/Diff21.py
'''
Given an int n, return the absolute difference between n and 21, except return
double the absolute difference if n is over 21.
diff21(19) → 2
diff21(10) → 11
diff21(21) → 0
'''

def diff21(n):
  if n > 21:
    return (n-21)*2
  return 21-n

print("diff21(19) -> " + str(diff21(19)))
print("diff21(10) -> " + str(diff21(10)))
print("diff21(21) -> " + str(diff21(21)))
print("diff21(30) -> " + str(diff21(30)))
print("diff21(0) -> " + str(diff21(0)))
